lib/rules/array.js

```diff
@@ -54,6 +54,14 @@ module.exports = function({ schema, messages }, path, context) {
 		`);
 	}
 
+	if (schema.unique === true) {
+		src.push(`
+			if(len > Array.from(new Set(value)).length) {
```

Comment:
The `Set` has a `size` property, no need to convert it back the array for length. Please update it.